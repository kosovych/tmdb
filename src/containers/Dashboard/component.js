import React from 'react'
import {
  Row, Col, Input, Pagination, Spin
} from 'antd'
import PropTypes from 'prop-types'

import MoviesList from 'Components/MoviesList'

const Dashboard = ({
  moviesData, movies, loading, onPageChange
}) => (
  <>
    <Row
      justify="center"
      gutter={{
        xs: 8, sm: 16, md: 24, lg: 22
      }}
    >
      <Col
        className="gutter-row"
        xs={{ span: 20 }}
        sm={{ span: 20 }}
        md={{ span: 14 }}
        lg={{ span: 12 }}
        xl={{ span: 10 }}
      >
        <Input.Search
          placeholder="Enter movie name"
          size="large"
          enterButton="Search"
          className="top-margin"
        />
      </Col>
    </Row>
    <div className="top-margin">
      <Row
        type="flex"
        gutter={16}
      >
        <Col
          className="gutter-row"
          span={20}
          offset={2}
        >
          <Row
            gutter={{
              xs: 8, sm: 16, md: 24, lg: 32
            }}
            justify={loading ? 'center' : 'start'}
          >
            {!loading && movies ? (
              <MoviesList
                moviesIDs={movies.items}
                moviesData={moviesData}
              />
            ) : <Col><Spin /></Col>
        }
          </Row>
        </Col>
      </Row>
      <Row
        type="flex"
        justify="center"
      >
        <Col>
          {movies && (
          <Pagination
            disabled={loading}
            defaultCurrent={movies.currentPage}
            onChange={onPageChange}
            className="pagination"
            showSizeChanger={false}
            total={movies.totalPages}
          />
          )}
        </Col>
      </Row>
    </div>
  </>
)

Dashboard.propTypes = {
  onPageChange: PropTypes.func.isRequired,
  moviesData: PropTypes.shape(),
  movies: PropTypes.shape({
    currentPage: PropTypes.number,
    items: PropTypes.arrayOf(PropTypes.number),
    totalPages: PropTypes.number
  }),
  loading: PropTypes.bool.isRequired
}

Dashboard.defaultProps = {
  moviesData: null,
  movies: null
}

export default Dashboard
