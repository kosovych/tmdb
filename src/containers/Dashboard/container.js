import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { getMovies as getMoviesAction } from 'Store/trendingMovies/actions'
import {
  moviesSelector,
  moviePagesSelector,
  movieErrorSelector,
  movieLoadingSelector,
  isBlankSelector
} from 'Store/trendingMovies/selectors'
import DashboardComponent from './component'

class Dashboard extends Component {
  componentDidMount = () => {
    const { getMovies } = this.props
    getMovies()
  }

  onPageChange = (page) => {
    const { getMovies } = this.props
    getMovies({ page })
  }

  render() {
    const {
      movies, loading, pagination: { currentPage, totalResults }, error, isBlank
    } = this.props
    return (
      <DashboardComponent
        movies={movies}
        currentPage={currentPage}
        totalResults={totalResults}
        loading={loading}
        onPageChange={this.onPageChange}
        error={error}
        isBlank={isBlank}
      />
    )
  }
}

Dashboard.propTypes = {
  getMovies: PropTypes.func.isRequired,
  loading: PropTypes.bool,
  pagination: PropTypes.shape({
    currentPage: PropTypes.number,
    totalResults: PropTypes.number
  }),
  movies: PropTypes.PropTypes.arrayOf(PropTypes.shape()),
  error: PropTypes.string,
  isBlank: PropTypes.bool
}

Dashboard.defaultProps = {
  loading: null,
  movies: null,
  pagination: {
    currentPage: null,
    totalResults: null
  },
  error: null,
  isBlank: false
}

export { Dashboard as DashboardContainer }

const mapStateToProps = state => ({
  loading: movieLoadingSelector(state),
  pagination: moviePagesSelector(state),
  movies: moviesSelector(state),
  error: movieErrorSelector(state),
  isBlank: isBlankSelector(state)
})

const mapDispatchToProps = {
  getMovies: getMoviesAction
}

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard)
