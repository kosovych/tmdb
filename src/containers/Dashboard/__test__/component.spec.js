import React from 'react'
import { shallow } from 'enzyme'
import Dashboard from '../component'

describe('Dashboard component', () => {
  const props = {
    movies: [
      {
        overview: 'The story of Usnavi, a...',
        backdropPath: '/1.jpg',
        posterPath: '/1.jpg',
        id: 467909,
        title: 'In the Heights',
        mediaType: 'movie'
      }, {
        overview: 'Determined to ...',
        backdropPath: '/1.jpg',
        id: 791373,
        posterPath: '/1.jpg',
        title: 'Justice League',
        mediaType: 'movie'
      }
    ],
    onPageChange: jest.fn(),
    currentPage: 1,
    totalResults: 1
  }

  describe('with movies', () => {
    it('should match snapshot', () => {
      const component = shallow(<Dashboard {...props} />)

      expect(component).toMatchSnapshot()
    })
  })

  describe('with loading', () => {
    const propsWithLoading = {
      movies: [],
      loading: true,
      onPageChange: jest.fn()
    }
    it('should match snapshot', () => {
      const component = shallow(<Dashboard {...propsWithLoading} />)
      expect(component).toMatchSnapshot()
    })
  })

  describe('with isBlank', () => {
    const propsWithIsBlank = {
      movies: [],
      isBlank: true,
      onPageChange: jest.fn()
    }
    it('should match snapshot', () => {
      const component = shallow(<Dashboard {...propsWithIsBlank} />)
      expect(component).toMatchSnapshot()
    })
  })

  describe('with error', () => {
    const propsWithError = {
      movies: [],
      error: 'Error',
      onPageChange: jest.fn()
    }
    it('should match snapshot', () => {
      const component = shallow(<Dashboard {...propsWithError} />)
      expect(component).toMatchSnapshot()
    })
  })
})
