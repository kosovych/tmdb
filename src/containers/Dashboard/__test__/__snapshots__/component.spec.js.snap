// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component with error should match snapshot 1`] = `
<Fragment>
  <Connect(WithFormik(Search)) />
  <div
    className="top-margin"
  >
    <Row
      gutter={16}
      type="flex"
    >
      <Col
        className="gutter-row"
        offset={2}
        span={20}
      >
        <Row
          gutter={
            Object {
              "lg": 32,
              "md": 24,
              "sm": 16,
              "xs": 8,
            }
          }
          justify="start"
        >
          <MoviesList
            movies={Array []}
          />
          <Col
            span={24}
          >
            <Alert
              message="Error"
              type="error"
            />
          </Col>
        </Row>
      </Col>
    </Row>
    <Row
      justify="center"
      type="flex"
    >
      <Col />
    </Row>
  </div>
</Fragment>
`;

exports[`Dashboard component with isBlank should match snapshot 1`] = `
<Fragment>
  <Connect(WithFormik(Search)) />
  <div
    className="top-margin"
  >
    <Row
      gutter={16}
      type="flex"
    >
      <Col
        className="gutter-row"
        offset={2}
        span={20}
      >
        <Row
          gutter={
            Object {
              "lg": 32,
              "md": 24,
              "sm": 16,
              "xs": 8,
            }
          }
          justify="start"
        >
          <MoviesList
            movies={Array []}
          />
          <Col
            span={24}
          >
            <Empty
              image={<Simple />}
            />
          </Col>
        </Row>
      </Col>
    </Row>
    <Row
      justify="center"
      type="flex"
    >
      <Col />
    </Row>
  </div>
</Fragment>
`;

exports[`Dashboard component with loading should match snapshot 1`] = `
<Fragment>
  <Connect(WithFormik(Search)) />
  <div
    className="top-margin"
  >
    <Row
      gutter={16}
      type="flex"
    >
      <Col
        className="gutter-row"
        offset={2}
        span={20}
      >
        <Row
          gutter={
            Object {
              "lg": 32,
              "md": 24,
              "sm": 16,
              "xs": 8,
            }
          }
          justify="center"
        >
          <Col>
            <Spin
              size="default"
              spinning={true}
              wrapperClassName=""
            />
          </Col>
        </Row>
      </Col>
    </Row>
    <Row
      justify="center"
      type="flex"
    >
      <Col />
    </Row>
  </div>
</Fragment>
`;

exports[`Dashboard component with movies should match snapshot 1`] = `
<Fragment>
  <Connect(WithFormik(Search)) />
  <div
    className="top-margin"
  >
    <Row
      gutter={16}
      type="flex"
    >
      <Col
        className="gutter-row"
        offset={2}
        span={20}
      >
        <Row
          gutter={
            Object {
              "lg": 32,
              "md": 24,
              "sm": 16,
              "xs": 8,
            }
          }
          justify="start"
        >
          <MoviesList
            movies={
              Array [
                Object {
                  "backdropPath": "/1.jpg",
                  "id": 467909,
                  "mediaType": "movie",
                  "overview": "The story of Usnavi, a...",
                  "posterPath": "/1.jpg",
                  "title": "In the Heights",
                },
                Object {
                  "backdropPath": "/1.jpg",
                  "id": 791373,
                  "mediaType": "movie",
                  "overview": "Determined to ...",
                  "posterPath": "/1.jpg",
                  "title": "Justice League",
                },
              ]
            }
          />
        </Row>
      </Col>
    </Row>
    <Row
      justify="center"
      type="flex"
    >
      <Col>
        <Pagination
          className="pagination"
          current={1}
          disabled={null}
          onChange={[MockFunction]}
          pageSize={20}
          showSizeChanger={false}
          total={1}
        />
      </Col>
    </Row>
  </div>
</Fragment>
`;
